packages/multichain/src/scope/validation.ts

```diff
@@ -0,0 +1,102 @@
+import { isCaipReference } from '@metamask/utils';
+
+import type {
+  ExternalScopeString,
+  ExternalScopeObject,
+  ExternalScopesObject,
+} from './types';
+import { parseScopeString } from './types';
+
+export const isValidScope = (
+  scopeString: ExternalScopeString,
+  scopeObject: ExternalScopeObject,
+): boolean => {
+  const { namespace, reference } = parseScopeString(scopeString);
+
+  if (!namespace && !reference) {
+    return false;
+  }
+
+  const {
+    references,
+    methods,
+    notifications,
+    accounts,
+    rpcDocuments,
+    rpcEndpoints,
+    ...restScopeObject
+  } = scopeObject;
+
+  if (!methods || !notifications) {
+    return false;
+  }
+
+  // These assume that the namespace has a notion of chainIds
```

Comment:
I'm getting a little confused about where we landed on this question - whether we should be enforcing that scopes always contain a reference (either in the scopeString itself or in a references array)

[CAIP-217 leaves the door open for scopes without references](https://github.com/ChainAgnostic/CAIPs/blob/9c0aeb3a90a91ad9ef705d916648fa7ce3db7cb8/CAIPs/caip-217.md?plain=1#L67)

Reply:
If we are enforcing this, then there are a couple of validation cases I think we're missing:
`ScopeString` contains namespace only and `references` is either absent or an empty array.

Reply:
I've added enforcement for these cases here: https://github.com/MetaMask/core/pull/4784/commits/664069be66cb1e751a9c7f6383ccbee0c2743657